browser/android/tab_features_android.cc

```diff
@@ -5,11 +5,18 @@
 
 #include "brave/browser/android/tab_features_android.h"
 
+#include <memory>
+
+#include "brave/browser/ai_chat/tab_data_web_contents_observer.h"
+#include "chrome/browser/android/tab_android.h"
 #include "chrome/browser/profiles/profile.h"
 #include "content/public/browser/web_contents.h"
 
 TabFeaturesAndroid::TabFeaturesAndroid(content::WebContents* web_contents,
                                        Profile* profile)
-    : TabFeaturesAndroid_Chromium(web_contents, profile) {}
+    : TabFeaturesAndroid_Chromium(web_contents, profile),
+      tab_data_observer_(std::make_unique<ai_chat::TabDataWebContentsObserver>(
+          TabAndroid::FromWebContents(web_contents)->GetAndroidId(),
```

Comment:
we can't use the same handle methods here? Are they not available on android?

Reply:
Unfortunately not - it's a bit of a mess. The `//chrome/browser/ui/tabs:impl` target is only included on non-android platforms, so the headers for the Handle methods are available but not the implementations.

I think upstream is probably going to need to bring these a bit closer together 'cause things are still pretty messy.

Additionally, the handle stuff is part of the `TabInterface` class, which isn't even used on Android :/ (as per that conversation you linked me about not sharing the Android/Desktop implementation).

Life would definitely be a bit easier if some of this stuff was shared.